src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IFieldReferenceExpression.cs

```diff
@@ -245,6 +245,66 @@ void M()
             VerifyOperationTreeAndDiagnosticsForTest<MemberAccessExpressionSyntax>(source, expectedOperationTree, expectedDiagnostics);
         }
 
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact]
+        public void IFieldReferenceExpression_IndexingFixed_Unmovable()
+        {
+            string source = @"
+
+unsafe struct S1
+{
+    public fixed int field[10];
+}
+
+unsafe class C
+{
+    void M()
+    {
+        S1 s = default;
+
+        /*<bind>*/ s.field[3] = 1; /*</bind>*/
+    }
+}
+";
+            string expectedOperationTree = @"
```

Comment:
How can this be the operation tree? There is no field access, no indexing and no assignment :-(

Reply:
I checked in the debugger, and all the nodes are represented. Perhaps this is how this stuff is visualized.  

---
In reply to: [176391814](https://github.com/dotnet/roslyn/pull/25660#discussion_r176391814) [](ancestors = 176391814)

Reply:
Also, the "unmovable" case was not affected by this feature. All we can require is that now "movable" and "unmovable" scenarios look the same. Which they do.

---
In reply to: [176553673](https://github.com/dotnet/roslyn/pull/25660#discussion_r176553673) [](ancestors = 176553673,176391814)

Reply:
Tagging @dotnet/analyzer-ioperation I think something is wrong here.

From my understanding the area delineated by `<bind>` comment should be a syntax node of the right kind (`VerifyOperationTreeAndDiagnosticsForTest<TSomeSyntaxNode>(...`.
I'd expect the printed IOperation tree to be an assignment, with left being a sub-tree and right being a literal 1.

Try `VerifyOperationTreeAndDiagnosticsForTest<AssignmentExpressionSyntax>(...` instead?

Reply:
Yes, that was the reason. Changed the syntax node kind and now we have whole trees.

---
In reply to: [176559103](https://github.com/dotnet/roslyn/pull/25660#discussion_r176559103) [](ancestors = 176559103)